builtin/nomad/jobspec/platform.go

```diff
@@ -74,6 +74,10 @@ func (p *Platform) resourceManager(log hclog.Logger, dcr *component.DeclaredReso
 	)
 }
 
+type nomadClient struct {
+	NomadClient *api.Client
+}
```

Comment:
After doing some digging, I see this comes from the former `releaser.go` file that was in here and both the platform and releaser shared this type. 

That said, the `builtin/nomad` tree now defines `getNomadClient` 4 times: 

```
[waypoint][main](4)$ ack getNomadClient builtin/nomad | grep func

builtin/nomad/task.go:286:func (p *TaskLauncher) getNomadClient() (*api.Client, error) {
builtin/nomad/platform.go:96:func (p *Platform) getNomadClient() (*api.Client, error) {
builtin/nomad/jobspec/releaser.go:73:func (r *Releaser) getNomadClient() (*nomadClient, error) {
builtin/nomad/jobspec/platform.go:79:func (p *Platform) getNomadClient() (*nomadClient, error) {
```

That itself might not be a problem but 2 of them return a `nomadClient` type that wraps the client, and the other 2 do not, and I don't understand why. Any thoughts on consolidating?

Reply:
Updated my original comment above to strike the "now" part; it seems before this change the `builtin/nomad` tree already defined the method 4 times :P

Reply:
@catsby after taking a look at this, I'm in favor of consolidating the two implementations of this in the `builtin/nomad` package, in `platform.go` and `task.go`. 

I'd like to leave the implementations in `builtin/nomad/jobspec` and `builtin/nomad/canary` separate though, just for the purpose of keeping the plugin packages more independent of each other (given that at some point we may move the built in plugins to their own repos altogether). Though it's not exactly the same situation, I intentionally did not update the `runnerinstall` package to use any utility methods under `builtin`, and likewise did not update anything in `builtin` to depend on the `installutil` package's helper methods, just to keep the plugin code more independent.

Regarding why the client is wrapped, when I initially implemented the resource manager framework for the jobspec & canary plugins earlier this year, I noticed that the k8s resource manager implementation wrapped the k8s client set in a struct. Albeit in that situation, the clientset, namespace and client config are provided in that struct (copied below for reference):

```go
// getClientset is a value provider for our resource manager and provides
// the connection information used by resources to interact with Kubernetes.
func (p *Platform) getClientset() (*clientsetInfo, error) {
	// Get our client
	clientSet, ns, config, err := Clientset(p.config.KubeconfigPath, p.config.Context)
	if err != nil {
		return nil, err
	}

	return &clientsetInfo{
		Clientset: clientSet,
		Namespace: ns,
		Config:    config,
	}, nil
}
```

Given that, since for Nomad we only need `*api.Client`, I think that `getNomadClient` in the jobspec and canary packages can be updated to return just that, instead of a struct wrapping it.

Reply:
Sounds rational, thanks 👍